broker-daemon/block-order-worker/index.spec.js

```diff
@@ -845,30 +848,28 @@ describe('BlockOrderWorker', () => {
 
     context('blockOrder is a limitOrder', async () => {
       context('blockOrder is a bid', async () => {
-        it('checks if the outbound balance in the engine is sufficient', async () => {
-          await worker.checkFundsAreSufficient(blockOrderStub)
-
-          expect(ltcEngine.isBalanceSufficient).to.have.been.calledOnce()
-          expect(ltcEngine.isBalanceSufficient).to.have.been.calledWith(outboundAddress, Big(outboundAmount).plus(activeOutboundAmount))
+        it('checks if the inbound/outbound balances in the engine are sufficient', async () => {
+          expect(await worker.checkFundsAreSufficient(blockOrderStub)).to.be.eql(undefined)
         })
 
-        it('checks if the inbound balance in the engine is sufficient', async () => {
-          await worker.checkFundsAreSufficient(blockOrderStub)
-
-          expect(btcEngine.isBalanceSufficient).to.have.been.calledOnce()
-          expect(btcEngine.isBalanceSufficient).to.have.been.calledWith(inboundAddress, Big(inboundAmount).plus(activeInboundAmount), { outbound: false })
+        it('throws if the outbound balance is greater than the amount they have in a single outbound channel', () => {
+          ltcEngine.getMaxChannelForAddress.withArgs(outboundAddress).resolves({ maxBalance: '10' })
+          return expect(worker.checkFundsAreSufficient(blockOrderStub)).to.be.rejectedWith('Insufficient funds in a single outbound LTC channel to create order')
         })
 
-        it('throws if the outbound balance is greater than the amount they have in the outbound channel', () => {
-          ltcEngine.isBalanceSufficient.withArgs(outboundAddress, Big(outboundAmount).plus(activeOutboundAmount)).resolves(false)
-
-          return expect(worker.checkFundsAreSufficient(blockOrderStub)).to.be.rejectedWith('Insufficient funds in outbound LTC channel to create order')
+        it('throws if the inbound balance is greater than the amount the relayer has in a single inbound channel', () => {
+          btcEngine.getMaxChannelForAddress.withArgs(inboundAddress).resolves({ maxBalance: '10' })
+          return expect(worker.checkFundsAreSufficient(blockOrderStub)).to.be.rejectedWith('Insufficient funds in a single inbound BTC channel to create order')
         })
 
-        it('throws the inbound balance is greater than the amount the relayer has in the inbound channel', () => {
-          btcEngine.isBalanceSufficient.withArgs(inboundAddress, Big(inboundAmount).plus(activeInboundAmount), { outbound: false }).resolves(false)
+        it('throws if the outbound balance is greater than the amount they have in all outbound channels', () => {
+          ltcEngine.getTotalBalanceForAddress.withArgs(outboundAddress).resolves('10')
```

Comment:
I think a more nuanced test case would be appropriate: the total balance is greater than the size of the order being placed, but less than the size of the order + existing active orders.

Reply:
Updated